File: 노현민/백준/b16926.cpp
#include <iostream>

using namespace std;

int main() {
	iostream::sync_with_stdio(0);
	cin.tie(0);
	cout.tie(0);

	int n, m, R;
	int map[301][301];


	cin >> n >> m >> R;


	for (int i = 0; i < n; ++i) {
		for (int j = 0; j < m; ++j) {
			cin >> map[i][j];
		}
	}

	int layers = min(n,m)/2;

	for (int i = 0; i < R; ++i) {
		for (int layer = 0; layer < layers; ++layer) {
			int tmp = map[layer][layer];
			int r = n - layer - 1, c = m - layer - 1;

			for (int j = layer; j < c;++j) {
				map[layer][j] = map[layer][j + 1];
			}
			for (int j = layer; j < r; ++j) {
				map[j][c] = map[j+1][c];
			}
			for (int j = c; j > layer; --j) {
				map[r][j] = map[r][j-1];
			}
			for (int j = r; j > layer; --j) {
				map[j][layer] = map[j-1][layer];
			}
			map[layer + 1][layer] = tmp;
		}
	}

	for (int i = 0; i < n; ++i) {
		for (int j = 0; j < m;++j) {
			cout << map[i][j] << " ";
		}
		cout << '\n';
	}


}

File: 김재환/src/월말평가/b16926.java
package 월말평가;

import java.util.*;
import java.io.*;

public class b16926 {
	static int n, m, r;
	static int[][] arr;
	public static void main(String[] args) throws Exception{
		System.setIn(new FileInputStream("Test4.txt"));
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		StringTokenizer st = new StringTokenizer(br.readLine());
		
		n = Integer.parseInt(st.nextToken());
		m = Integer.parseInt(st.nextToken());
		r = Integer.parseInt(st.nextToken());
		arr= new int[n][m];
		
		for(int i=0; i<n; i++) {
			st = new StringTokenizer(br.readLine());
			for(int j=0; j<m; j++) {
				arr[i][j] = Integer.parseInt(st.nextToken());
			}
		}
		
		rotate();
	}
	
	static void rotate() {
		int layers = Math.min(n, m)/2;
		
		for(int layer =0; layer<layers; layer++) {
			List<Integer> tmp = new ArrayList<>();
			
			
			for(int j=layer; j<m-layer; j++) tmp.add(arr[layer][j]);
			for(int i=layer+1; i<n-layer-1; i++) tmp.add(arr[i][m-layer-1]);
			for(int j=m-layer-1; j>layer; j--) tmp.add(arr[n-layer-1][j]);
			for(int i=n-layer-1; i>layer; i--) tmp.add(arr[i][layer]);

			int len = tmp.size();
			int rot = r%len;
			
			List<Integer> rotate = new ArrayList<>();
			
			for(int i=0; i<len; i++) {
				rotate.add(tmp.get((i+rot)%len));
			}
			
			int idx = 0;
			for(int j=layer; j<m-layer; j++) arr[layer][j]=rotate.get(idx++);
			for(int i=layer+1; i<n-layer-1; i++) arr[i][m-layer-1] = rotate.get(idx++);
			for(int j=m-layer-1; j>layer; j--) arr[n-layer-1][j] = rotate.get(idx++);
			for(int i=n-layer-1; i>layer; i--) arr[i][layer] = rotate.get(idx++);
		}
		
		
		StringBuilder sb = new StringBuilder();
		for(int[] row: arr) {
			for(int num : row) {
				sb.append(num).append(" ");
			}
			sb.append('\n');
		}
		
		System.out.print(sb);
	}
}

